pulp_service/pulp_service/app/authorization.py

```diff
@@ -44,12 +46,28 @@ def has_permission(self, request, view):
         # Anyone can create a domain
         if request.META['PATH_INFO'].endswith('/default/api/v3/domains/') and request.META['REQUEST_METHOD'] == 'POST':
             org_id_var.set(org_id)
+            user_id_var.set(request.user.pk)
             return True
-        # User has permission if the org_id matches the domain's org_id
+        # User has permission if the org_id matches the domain's org_id or the user has group perms
         else:
-            domain_name = request.META['PATH_INFO'].removeprefix(settings.API_ROOT).split("/")[0]
-            try:
-                DomainOrg.objects.get(domain__name=domain_name, org_id=org_id)
-                return True
-            except DomainOrg.DoesNotExist:
-                return False
+            checker = DomainPermissionChecker(request, org_id)
+            return checker.has_permissions()
+
+
+class DomainPermissionChecker:
+    def __init__(self, request, org_id):
+        self.user = request.user
+        self.domain_name = request.META['PATH_INFO'].removeprefix(settings.API_ROOT).split("/")[0]
+        self.org_id = org_id
+
+    def has_permissions(self):
+        return self.has_org_id_perms() or self.has_user_perms() or self.has_group_perms()
```

Comment:
Does the order matter when checking for the user-domain permissions? Should we prioritize one over the other?

Reply:
I suspect we want the lowest latency with user or group permissions (put those first) and then ord_id permissions last because that's likely a human doing it.

This I think is fine for now, but one future idea would be to produce one query that checks all 3 things at once and brings all the data across the cursor once instead of going back to the DB 3 times.